solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,522 @@
+    {
+        // initialize both the source and the target tokens
+        IERC20[] memory reserveTokens = new IERC20[](2);
+        reserveTokens[0] = _networkToken;
+        reserveTokens[1] = token;
+
+        // make sure to only look up for 50/50 converters
+        uint32[] memory standardReserveWeights = new uint32[](2);
+        standardReserveWeights[0] = STANDARD_POOL_RESERVE_WEIGHT;
+        standardReserveWeights[1] = STANDARD_POOL_RESERVE_WEIGHT;
+
+        // find the standard pool converter between the specified token and the network token
+        IConverterAnchor anchor =
+            converterRegistry.getLiquidityPoolByConfig(
+                STANDARD_POOL_CONVERTER_TYPE,
+                reserveTokens,
+                standardReserveWeights
+            );
+        require(address(anchor) != address(0), "ERR_INVALID_RESERVE_TOKEN");
+
+        return anchor;
+    }
+
+    /**
+     * @dev ensures that the network is able to pull the tokens from this contact
+     *
+     * @param token the source token
+     * @param network the address of the network contract
+     * @param amount the token amount to approve
+     */
+    function ensureAllowance(
+        IERC20 token,
+        BancorNetwork network,
+        uint256 amount
+    ) private {
+        address networkAddress = address(network);
+        uint256 allowance = token.allowance(address(this), networkAddress);
+        if (allowance < amount) {
+            if (allowance > 0) {
+                token.safeApprove(networkAddress, 0);
+            }
+            token.safeApprove(networkAddress, amount);
+        }
+    }
+
+    /**
+     * @dev returns the converter registry
+     *
+     * @return the converter registry
+     */
+    function converterRegistry() private view returns (IConverterRegistry) {
+        return IConverterRegistry(addressOf(CONVERTER_REGISTRY));
+    }
+
+    /**
+     * @dev returns the network contract
+     *
+     * @return the network contract
+     */
+    function bancorNetwork() private view returns (BancorNetwork) {
+        return BancorNetwork(payable(addressOf(BANCOR_NETWORK)));
+    }
+
+    /**
+     * @dev returns the network settings contract
+     *
+     * @return the network settings contract
+     */
+    function networkSetting() private view returns (INetworkSettings) {
+        return INetworkSettings(addressOf(NETWORK_SETTINGS));
+    }
+
+    /**
+     * @dev returns the network fee wallet
+     *
+     * @return the network fee wallet
+     */
+    function networkFeeWallet() private view returns (ITokenHolder) {
+        return ITokenHolder(networkSetting().networkFeeWallet());
+    }
+
+    /**
+     * @dev returns whether the provided list of tokens include duplicates
+     *
+     * @param tokens the list of tokens to check
+     *
+     * @return whether the provided list of tokens include duplicates
+     */
+    function hasDuplicates(IERC20[] calldata tokens) private pure returns (bool) {
```

Comment:
Perhaps rename this function to `invalidTokenList` and also check for zero addresses?

Reply:
This method is only used in `availableVortex`, so the name would be misleading. I'd prefer to keep it. It's more descriptive.